src/psbtv2.ts

```diff
@@ -0,0 +1,910 @@
+  PSBT_IN_PREVIOUS_TXID = '0e',
+  PSBT_IN_OUTPUT_INDEX = '0f',
+  PSBT_IN_SEQUENCE = '10',
+  PSBT_IN_REQUIRED_TIME_LOCKTIME = '11',
+  PSBT_IN_REQUIRED_HEIGHT_LOCKTIME = '12',
+  PSBT_IN_TAP_KEY_SIG = '13',
+  PSBT_IN_TAP_SCRIPT_SIG = '14',
+  PSBT_IN_TAP_LEAF_SCRIPT = '15',
+  PSBT_IN_TAP_BIP32_DERIVATION = '16',
+  PSBT_IN_TAP_INTERNAL_KEY = '17',
+  PSBT_IN_TAP_MERKLE_ROOT = '18',
+  PSBT_IN_PROPRIETARY = 'fc',
+  
+  PSBT_OUT_REDEEM_SCRIPT = '00',
+  PSBT_OUT_WITNESS_SCRIPT = '01',
+  PSBT_OUT_BIP32_DERIVATION = '02',
+  PSBT_OUT_AMOUNT = '03',
+  PSBT_OUT_SCRIPT = '04',
+  PSBT_OUT_TAP_INTERNAL_KEY = '05',
+  PSBT_OUT_TAP_TREE = '06',
+  PSBT_OUT_TAP_BIP32_DERIVATION = '07',
+  PSBT_OUT_PROPRIETARY = 'fc',
+}
+
+
+// Provided to friendly-format the PSBT_GLOBAL_TX_MODIFIABLE bitmask from
+// PsbtV2.PSBT_GLOBAL_TX_MODIFIABLE which returns PsbtGlobalTxModifiableBits[]. 
+enum PsbtGlobalTxModifiableBits {
+  INPUTS = 'INPUTS', // 0b000000001
+  OUTPUTS = 'OUTPUTS', // 0b000000010
+  SIGHASH_SINGLE = 'SIGHASH_SINGLE', // 0b000000100
+}
+
+
+/* 
+Global Constants
+ */
+
+const PSBT_MAGIC_BYTES = Buffer.from([0x70, 0x73, 0x62, 0x74, 0xff]);
+const PSBT_MAP_SEPARATOR = Buffer.from([0x00]);
+const BIP_32_NODE_REGEX = /(\/[0-9]+'?)/ig;
+const BIP_32_HARDENING_OFFSET = 0x80000000;
+const ERROR_PSBT_NOT_VALID = Error("Not a valid psbt");
+
+
+/* 
+Helper Functions
+*/
+
+// Return psbt as Buffer.
+function bufferize (psbt:string|Buffer):Buffer {
+  let buf:Buffer|undefined;
+
+  if (Buffer.isBuffer(psbt)) {
+    buf = psbt;
+  } else if (typeof psbt === 'string' && (/^([0-9A-Fa-f][0-9A-Fa-f])+$/).test(psbt)) {
+    // Is a hex string
+    buf = Buffer.from(psbt, 'hex');
+  } else if (
+    typeof psbt === 'string' && 
+    (/^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$/).test(psbt)
+  ) {
+    // Is a base64 string
+    buf = Buffer.from(psbt, 'base64');
+  } else {
+    throw ERROR_PSBT_NOT_VALID;
+  }
+
+  return buf;
+}
+
+
+// Some keytypes have keydata which allows for multiple unique keys of the same
+// keytype. Getters which return values from these keys should search and return
+// values from all keys of that keytype. This function matches on the first byte
+// of each key string (hex encoded) and returns all values associated with those
+// keys as an array of string (hex encoded) values.
+function getNonUniqueKeyTypeValues (
+  maps:Map<Key, Value>|Map<Key, Value>[], 
+  keytype:KeyType
+) {
+  if (Array.isArray(maps)) {
+    // It's a set of input or output maps, so recursively check each map and set
+    // values.
+    const values:({ key:string, value:string|null})[][] = [];
+
+    for (const map of maps) {
+      values.push(
+        // There is probably a better way to type this
+        getNonUniqueKeyTypeValues(map, keytype) as { key:string, value:string|null }[]
+      );
+    }
+
+    return values;
+  }
+  
+  const map = maps;
+  const vals:{ key:string, value:string|null }[] = [];
```

Comment:
this type is being repeated. will think more on if there's a better way to do the typing above but at the very least this could be pulled out as something like 

```typescript
type UniqueKeyTypeVals = { key:string, value:string|null }[]
```

or something